src/model/src/InertialParametersHelpers.cpp

```diff
@@ -113,24 +119,33 @@ bool BBFromExternalShape(ExternalMesh* extMesh, Box& box)
     if (pScene)
     {
         // Extract vector of vertices
-        double scale = 1.0;
+        double scale = 1;
         std::vector<aiVector3D> vertexVector;
         buildMesh(pScene, pScene->mRootNode, scale, vertexVector);
 
         if (vertexVector.size() == 0)
         {
             return false;
         }
-
+        
+        
+        double scaling_factor=extMesh->scale.getVal(0);
+        for(size_t i=0; i < vertexVector.size(); i++)
+        {
+            vertexVector[i]*=scaling_factor;
+        }
```

Comment:
Even if I am not enthusiastic about it, in the URDF the scale attribute is a three valued parameters (see https://github.com/PR2/pr2_common/blob/6f96993793230c81737478729c4e324c3574424a/pr2_description/urdf/sensors/kinect_camera.urdf.xacro#L25). For this reason, we should not just consider the first value of the scale, but scale every vertex for the different scaling value, even if in this case the scaling vector is the same for all axis.

Reply:
See https://answers.ros.org/question/273434/how-to-resize-meshes-in-urdf/ for a related ROS Answer question.